src/sparseml/exporters/kv_cache_injector.py

```diff
@@ -80,63 +89,53 @@ def __init__(
         :param model_path: The path to the directory containing the model.
         :param inplace: If True, the model will be modified in place.
             If False, a copy of the model will be made and modified.
+        :param kwargs: (Optionally) the parameters for the KV cache injection
+            if no `model_path` is provided.
         """
         self.inplace = inplace
-        if model_path:
-            self.config = self.get_config(model_path)
-            if not self.config["model_type"] in _SUPPORTED_ARCHITECTURES:
-                raise ValueError(
-                    f"Model type {self.config.model_type} is currently not supported. "
-                    f"Supported model types: {_SUPPORTED_ARCHITECTURES}."
-                    f"Proceeding with transformation, but may require additional "
-                    f"customization..."
-                )
-
-            num_attention_heads = self.config["num_attention_heads"]
-            hidden_size_kv_cache = self.config["hidden_size"] // num_attention_heads
+
+        if model_path is not None:
+            # get the parameters from the config
+            self.config = get_kv_cache_config(model_path)
```

Comment:
what happens if config is `None` looks like we crash (after a warning) instead of defaulting?

Reply:
You mean when there are no `kwargs` and `config = None` right? Yes, this is a corner case that I should cover, good point.

Reply:
Second thought, this is already covered. If config is `None` we will end up raising an error. I think this makes sense, defaulting here is not a good idea IMO